### ISSUE TITLE
Checkpoint issue

### ISSUE THREAD

**Issue by juliencelia** [SmileiPIC/Smilei#333]
Dear Smilei team,

I ran a big 2D simulation (800MPI/16PopenMP) on ROME. The Smilei version is v4.4-706-gb5c12a5a-master
I have a new issue with checkpoints at 2nd launch:
```
 [ERROR](0) src/Tools/PyTools.h:154 (checkPyError) Exception: ERROR in the namelist: cannot find valid restart files for processor 240
                restart_dir = '/ccc/scratch/cont003/ra5390/bonvalej/tir_direct/checkpoint000'
                restart_number = None
```
And iti s right in my checkpoints/ folder, I have 27 folders (from 00 to 26). Only 00/ folder is not empty with just one file dump-00000-0000000000.h5. This file is 2.1Go. I stop simulation and write dumps at 1300min and the simulation duration is 86000sec...

Here are the files:

[out.txt](https://github.com/SmileiPIC/Smilei/files/5530493/out.txt)
[BNH2dtest.txt](https://github.com/SmileiPIC/Smilei/files/5530494/BNH2dtest.txt)
[job_sub.txt](https://github.com/SmileiPIC/Smilei/files/5530495/job_sub.txt)

Thanks

**Comment by mccoys** [comment]
@iltommi There should be three zeroes with 800 MPI, right ? There are only two zeroes it seems. Could it be why it did not output ? Where there any changes between 4.4 and 4.5 ?

**Comment by juliencelia** [comment]
Hi,

The "dump_step" works fine... I had an issue with the "dump_minute" method.
If that can help you.

Cheers

Julien

**Comment by iltommi** [comment]
Sorry I just noticed my mention...

@mccoys : indeed we should generate 2digits dir, looking here: https://github.com/SmileiPIC/Smilei/blob/master/src/Python/pycontrol.py#L25
```python
smilei_mpi_size=800
file_grouping=30
ngroups = int((smilei_mpi_size-1)/file_grouping + 1)
ngroups_chars = int(math.log10(ngroups))+1
```
gives 
```python
ngroups_chars=2
ngroups=27
```

and looking later in the same file https://github.com/SmileiPIC/Smilei/blob/master/src/Python/pycontrol.py#L50
We're globbing a pattern like `restart_dir+"/checkpoints/*/dump-*-*.h5"` and the number of digits is only important for the tree creation, not for the restart

@juliencelia:
What I don't understand is that it looks like the first simulation did not create the dump files, right?
So the second is not be able to restart, right? 
Because the output file is truncated just after:
```
[...]
    79000/142811     3.1260e+03     7.7325e+04   (  9.0558e+02 )
 Reached time limit : 1300.012 minutes. Dump timestep : 79667
 Step 79667 : DUMP fields and particles 0
```

so we don't have a clean exit.

Do you have any error file?

**Comment by juliencelia** [comment]
The err file was empty...
I wont try to reproduce the same error because the simulation takes 300k hrs cpu.
Perhaps it was a bug of writing by cluster machine.

I will run some 3D and 2D AM simulations and I will test the next time...

Thanks

Julien

**Comment by beck-llr** [comment]
Julien,
we've been hvaving this kind of error for a while with very large 3D simulations. After several correction attempts, it looks like it is still not 100% robust yet. It looks like it depends on the filesystem of the super computer you're using.
I advise using `dump_step` for XXL simulations.

**Comment by iltommi** [comment]
@juliencelia  you're running a 4.4 version of smilei with commit is #gb5c12a5a which I cannot find in the history.
It might be that we fixed this after #223 was corrected and you're using an older version?
when did you pull from github?

**Comment by mccoys** [comment]
@iltommi https://github.com/SmileiPIC/Smilei/commit/b5c12a5ada3b0833cc89ed09eed94476699a9923

**Comment by iltommi** [comment]
ah yes! sorry, nevermind... @juliencelia forget my comment

**Comment by juliencelia** [comment]
It is strange I pulled from Git one month ago because I reinstall SMILEI on Rome AMDmachine...
I can retry to repull it.

**Comment by juliencelia** [comment]
No problem with dump_step with large simulations. It works.
